<!--
 * @Author: albertxin albert_xin@qq.com
 * @Date: 2024-07-24 10:46:27
 * @LastEditors: albertxin albert_xin@qq.com
 * @LastEditTime: 2024-07-26 09:36:05
 * @FilePath: /shixinblog/blog/2024-07-23-记录terminal.md
 * Copyright (c) albertxin by albert_xin@qq.com, All Rights Reserved. 
-->
# 记录terminal
```bash
# 对基因组建立索引
bwa-mem2 index /path/to/reference.fa
```

## 对基因组建立faidx索引

```bash
# 对基因组建立faidx索引
samtools faidx /path/to/reference.fa
```

```bash
# 利用bwa-mem2 进行比对


Csa=yourgenome.fa
$bwa2 mem -M -t 5 -R '@RG\tID:CSA\tLB:CSA\tPL:ILLUMINA\tSM:$Csa' $Csa yourfastqfile_R1.fastq.gz yourfastqfile_R2.fastq.gz | samtools sort -O bam -@ 4 -o Csa_reseq.bam - && samtools index Csa_reseq.bam

Cfi=yourgenome.fa
$bwa2 mem -M -t 5 -R '@RG\tID:CSA\tLB:CSA\tPL:ILLUMINA\tSM:$Cfi' $Cfi yourfastqfile_R1.fastq.gz yourfastqfile_R2.fastq.gz | samtools sort -O bam -@ 4 -o Cfi_reseq.bam - && samtools index Cfi_reseq.bam
```
## 利用awk提取染色体id

```bash
awk '/>ch/ {print}' Cucumber_genome_CLv4.fa |awk -F '>' '{if (NF>1) print $2}' >chr_id.txt
awk '/>ch/ {print}' Cfi_genome_from_JZG.fa |awk -F '>' '{if (NF>1) print $2}' >chr_id.txt
```

## 利用gatk对比对结果降重

```bash
gatk=/share/nas6/zhouxy/biosoft/GATK/gatk-4.2.6.0/gatk
mkdir temp
ls *bam|cut -d"." -f 1 |while read bam;do $gatk  --java-options -Xmx50g MarkDuplicates --TMP_DIR ./temp --INPUT $bam.bam --OUTPUT ../bam_dedup/$bam.dedup.bam --METRICS_FILE ../bam_dedup/$bam.dedup.metrics --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 1000 --CREATE_INDEX TRUE --REMOVE_DUPLICATES TRUE && samtools index ../bam_dedup/$bam.dedup.bam ;done
```

## 利用samtools统计结果

```bash
ls ../raw_bam/*bam |cut -d"/" -f 3|cut -d"." -f 1 |while read rawbam;do samtools stats ../raw_bam/$rawbam.bam >./bcfiles/$rawbam.bc;done

ls ../bam_dedup/*bam |cut -d"/" -f 3|cut -d"." -f1|while read id;do samtools depth ../bam_dedup/$id.dedup.bam >./depth2calc/$id.depth;done
samtools flagstat -@ 10 Csa_reseq.dedup.bam > Csa_reseq.dedup.flagstat
samtools flagstat -@ 10 Cfi_reseq.dedup.bam > Cfi_reseq.dedup.flagstat

ls ./depth2calc/*depth|cut -d"/" -f 3|cut -d"." -f1|while read id ;do depth_stat_windows -i ./depth2calc/$id.depth -o ./depth2plot/$id.depth.100Kb.fordraw.xls -w 100000;done
Rs=/share/nas6/zhouxy/biosoft/R/3.6.2/bin/Rscript
scirpt_Bin=/share/nas6/pub/pipline/dna-seq/bsa-seq/current/

ls ./depth2plot/*xls|cut -d"/" -f 3|cut -d"." -f1|while read id;do $Rs $scirpt_Bin/script/01bam/genomeCoveragehorizontalArea.R --infile ./depth2plot/$id.
```
