<!--
 * @Author: albertxin albert_xin@qq.com
 * @Date: 2024-07-29 08:38:14
 * @LastEditors: albertxin albert_xin@qq.com
 * @LastEditTime: 2024-07-29 18:00:08
 * @FilePath: /shixinblog/blog/2024-07-29-记录terminal.md
 * Copyright (c) albertxin by albert_xin@qq.com, All Rights Reserved. 
-->
# 记录terminal
---

## 利hisat2比对得后续，利用的fastq文件进行比对

### 利用blastn将比对上的Reads比对到黄瓜的转录组上
```bash
blastn -query ../02.unmapped2Cfi_hisat2/Cfi_rec_mapped.fa -db ../../rna-seq/03.rmdup.balstndb/Csa/CsaCsa_rmdup_rnaseq -out Cfi_rec_mapped2csacsa_rnaseq.blastn.out -outfmt 6 -num_threads 40 -max_target_seqs 10

# 不要把hits（-max_target_seqs）设置的太高，不然输出文件会很大，影响速度和浪费磁盘空间
```

### 将不完全匹配的序列提取出来
```bash
python /share/nas2/zhushixin/python_test/blastxml_extract/blastn_fmt6_filter_unperfect.py Cfi_rec_mapped2csacsa_rnaseq.blastn.out unperfect_match_mismatch_id.txt
```
### 根据id将fasta序列提取出来
```bash
seqkit grep ../02.unmapped2Cfi_hisat2/Cfi_rec_mapped.fa -f Cfi_rec_maped2csacsa_unperfectmatch_id.txt > unperfect_match.fa
```

### 利用blastn将unperfect_match.fa比对到南瓜cds上
```bash
tblastx -query unperfect_match_mismatch.fa -db ../../re-seq/re_con_genome/blastn_index/Cfi_cds -out unperfe
ct_match2csacsa_2Cfi_cds.blastn.out -outfmt 6 -num_threads 40 -max_target_seqs 10
```



---
## 利用tophat2比对fastq格式

### 利用Bowtie2建立索引

```bash
bowtie2-build --threads 10 Cfi_genome_reco_JZG.fa Cfi_genome_reco_JZG
bowtie2-build --threads 20 -f Cucumber_genome_reco.fa Cucumber_genome_reco
```
### 利用tophat2 异源嫁接的转录组数据比对到黄瓜重建基因组上

```bash
tophat2 -p 30 --read-edit-dist 3 ../../re-seq/re_con_genome/tophat_index/Csa/Cucumber_genome_reco  ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R2.fq.gz 
```

### 将未必对上的序列转换成fastq

```bash 
bedtools bamtofastq -i unmapped.bam -fq /dev/stdout |pigz > ../../02.unmapped2Cfi_tophat2_fq/tophatfq_csag_unmmaped.fq.gz
```

### 利用tophat2 参数为allow 1 mismatch 进行比对

```bash
tophat2 -p 30 -N 1 --read-gap-length 1 --read-edit-dist 1 ../../re-seq/re_con_genome/tophat_index/Cfi/Cfi_genome_reco_JZG tophatfq_csag_unmmaped.fq.gz
```

### 将比对上的文件转为fasta格式

```bash
samtools fasta accepted_hits.bam > ../../03.mapped2csacsarnaseq_tophat_fq/tophat_fq_2Cfi_mapped.fa
```




### 利用blastn将比对上的Reads比对到黄瓜的转录组上

```bash
blastn -query  tophat_fq_2Cfi_mapped.fa -db ../../rna-seq/03.rmdup.balstndb/Csa/CsaCsa_rmdup_rnaseq -out tophat_fq_2Cfi_mapped2csacsa_rnaseq.blastn.out -outfmt 6 -num_threads 40 -max_target_seqs 10
```
### 将不完全匹配的序列提取出来

```bash
python /share/nas2/zhushixin/python_test/blastxml_extract/blstn_xml_filter.py tophat_fq_2Cfi_mapped2csacsa_rnaseq.blastn.out tophat_fq_csacsa_mappedid.txt
```


### 利用tophat2 参数为allow 3 mismatch 进行比对

```bash

tophat2 -p 30 --read-edit-dist 3 -o tophat_out_al3  ../../re-seq/re_con_genome/to
phat_index/Cfi/Cfi_genome_reco_JZG tophatfq_csag_unmmaped.fq.gz
```



### 将不完全匹配Reads比对到Cfi的cds上

```bash 
tblastx -query ../03.mapped2csacsarnaseq_tophat_fq/tophat_fq_2Cfi_mapped.fa -db ../../re-seq/re_con_genome/blastn_index/Cfi_cds -out tophat_fq_csacsa_mapped2Cficds.out -outfmt 6 -evalue 1e-40 -num_threads 40 -max_target_seqs 10


## 比对到重新提取的Cfi的cds上
tblastx -query ../03.mapped2csacsarnaseq_tophat_fq/tophat_fq_2Cfi_mapped.fa -db ../../re-seq/re_con_genome/blastn_index/rc_cds/Cfi_cds_rc -out  tophat_fq_csacsa_mapped2Cficds_rc.out -outfmt 6 -evalue 1e-40 -num_threads 40 -max_target_seqs 10

```

## 利用bowtie2进行比对


```bash
bowtie2 -x ../../re-seq/re_con_genome/tophat_index/Csa/Cucumber_genome_reco -p 30  -1 ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz -2 ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R2.fq.gz  -S CsaCfi2Csa_rec.sam
```

### 将sam格式文件转换成bam格式文件

```bash 
samtools view -bS CsaCfi2Csa_rec.sam > CsaCfi2Csa_rec.bam
```


### 利用samtools将未比对上的序列提取出来

```bash 
```


## 新处理bcf文件
### bcftools过滤掉低深度的variant

```bash
bcftools filter -i 'DP > 70 ' -o filtered_Cfi_70.vcf Cfi_reseq.vcf.gz
bcftools filter -i 'DP > 70 ' -o filtered_Csa_70.vcf Csa_reseq.vcf.gz

```

### 对过滤后的vcf文件建立索引

```bash
bgzip filtered_Cfi_70.vcf && tabix -p vcf filtered_Cfi_70.vcf.gz
bgzip filtered_Csa_70.vcf && tabix -p vcf filtered_Csa_70.vcf.gz
```

### 利用bcftools进行consensus序列的生成

```bash
bcftools consensus -f ../../../genome/Cfi/Cfi_genome_from_JZG.fa ../../Cfi_out/bcf_call/filtered_Cfi_70.vcf.gz -o ./Cfi_genome_rc.fa
bcftools consensus -f ../../../genome/Csa/Cucumber_genome_CLv4.fa filtered_Csa_70.vcf.gz -o ../../rc_genome/Csa_rc.fa
```
### 利用bowtie2-build建立索引

```bash
bowtie2-build -f Cfi_genome_rc.fa ./bowtie2-index/Cfi_genome_rc
```

### 利用tophat2进行比对
```bash
tophat2 -p 30 --read-edit-dist 3 ../../re-seq/rc_genome/Csa/bowtie2-index/Csa_genome_rc ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R2.fq.gz
```
