<!--
 * @Author: albertxin albert_xin@qq.com
 * @Date: 2024-07-30 08:42:19
 * @LastEditors: albertxin albert_xin@qq.com
 * @LastEditTime: 2024-07-31 08:23:55
 * @FilePath: /shixinblog/blog/2024-07-30-记录terminal.md
 * Copyright (c) albertxin by albert_xin@qq.com, All Rights Reserved. 
-->
## 新处理bcf文件
### bcftools过滤掉低深度的variant

```bash
bcftools filter -i 'DP > 70 ' -o filtered_Cfi_70.vcf Cfi_reseq.vcf.gz
bcftools filter -i 'DP > 70 ' -o filtered_Csa_70.vcf Csa_reseq.vcf.gz

```

### 对过滤后的vcf文件建立索引

```bash
bgzip filtered_Cfi_70.vcf && tabix -p vcf filtered_Cfi_70.vcf.gz
bgzip filtered_Csa_70.vcf && tabix -p vcf filtered_Csa_70.vcf.gz
```

### 利用bcftools进行consensus序列的生成

```bash
bcftools consensus -f ../../../genome/Cfi/Cfi_genome_from_JZG.fa ../../Cfi_out/bcf_call/filtered_Cfi_70.vcf.gz -o ./Cfi_genome_rc.fa
bcftools consensus -f ../../../genome/Csa/Cucumber_genome_CLv4.fa filtered_Csa_70.vcf.gz -o ../../rc_genome/Csa_rc.fa
```
### 利用bowtie2-build建立索引

```bash
bowtie2-build -f Cfi_genome_rc.fa ./bowtie2-index/Cfi_genome_rc
```

### 利用tophat2进行比对
```bash
tophat2 -p 30 --read-edit-dist 3 ../../re-seq/rc_genome/Csa/bowtie2-index/Csa_genome_rc ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R2.fq.gz
```


### 提取比对结果中未比对上的read是
```bash
samtools fastq ../01.fq2Csa_rc/tophat_out/unmapped.bam  >unmapped.fq
```

### 利用tophat2将未比对上的Reads比对到Cfi-rc上
```bash
tophat2 -p 30 -N 1 --read-gap-length 1 --read-edit-dist 1 ../../re-seq/rc_genome/Cfi/bowtie2-index/Cfi_genome_rc ./unmapped.fq
```
### 将比对结果转化为fasta
```bash
samtools fastq accepted_hits.bam > ../../03.mapped2csacsarnaseq
/mapped.fa
```

### 利用blastn将比对上的序列比对到csacsarnaseq上去
```bash
blastn -query ./tophat_out/accepted_hits.bam -db ../../re-seq/rc_genome/Csa/csacsarnaseq/csacsarnaseq.fa -outfmt 6 -out ./tophat_out/csacsarnaseq.out
```
### 利用gffread将cds提取出来
```bash
gffread -w ./Csa_genome_rc_cds.fa -g Csa_genome_rc.fa ../../re_con_genome/Csa/Cucumber_genome_CLv4.gff3 
```

### 利用BLAST工具重新构建blastdb
```bash
makeblastdb -in Cfi_genome_rc_cds.fa -dbtype nucl -out ./blastndb/Cfi_genome_rc_cds
```

### 利用blastn将比对上的序列比对到CsaCsarnaseq上去转录组数据
```bash
blastn -query ./mapped.fa -db ../../rna-seq/03.rmdup.balstndb/Csa/CsaCsa_rmdup_rnaseq -out ./mapped2csacsarnaseq.out  -outfmt 6 -num_threads 30 -max_target_seqs 5
```
### 根据id将比对结果转化为fasta
```bash
seqkit grep ./mapped.fa -f unperfectmp_id.txt > ../04.unpfct2Cficds/unperfect_mp.fa

```
### 利用tblastx将不完全比对的序列比对到Cficds上去
```bash
tblastx -query ./unperfect_mp.fa -db ../../re-seq/rc_genome/Cfi/blastndb/Cfi_genome_rc_cds -out unperfect_mp2Cfi_cds.tblastx.out -outfmt 6 -num_threads 40 -max_target_seqs 10  -evalue 1e-40

```


## 利用tophat2默认参数进行比对

### 将CsaCfi转录组数据比对到CsaCsa上
```bash
tophat2 -p 30 ../../re-seq/rc_genome/Csa/bowtie2-index/Csa_genome_rc.fa ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz  ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz
```
