<!--
 * @Author: albertxin albert_xin@qq.com
 * @Date: 2024-07-26 08:31:37
 * @LastEditors: albertxin albert_xin@qq.com
 * @LastEditTime: 2024-07-29 08:39:40
 * @FilePath: /shixinblog/blog/2024-07-26-记录terminal.md
 * Copyright (c) albertxin by albert_xin@qq.com, All Rights Reserved. 
-->
## 利用tophat2比对fastq格式

### 利用Bowtie2建立索引

```bash
bowtie2-build --threads 10 Cfi_genome_reco_JZG.fa Cfi_genome_reco_JZG
bowtie2-build --threads 20 -f Cucumber_genome_reco.fa Cucumber_genome_reco
```
### 利用tophat2 异源嫁接的转录组数据比对到黄瓜上

```bash
tophat2 -p 30 --read-edit-dist 3 ../../re-seq/re_con_genome/tophat_index/Csa/Cucumber_genome_reco  ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R2.fq.gz 
```

### 将未必对上的序列转换成fastq

```bash 

bedtools bamtofastq -i unmapped.bam -fq /dev/stdout -fq2 /dev/stdout | pigz > ../../02.unmapped2Cfi_tophat2_fa/tophatfa_csag_unmapped.fq.gz
```


## 利用tophat2比对fasta格式的转录组序列


### 利用fasta比对
```bash
tophat2 -p 20 --b2-f  --read-edit-dist 3 ../../re-seq/re_con_genome/tophat_index/Csa/Cucumber_genome_reco ../../rna-seq/01-merged-fa/CsaCfi_100bp.fa.gz 
```
### 将未必对上的序列转换成fastq

```bash 

bedtools bamtofastq -i unmapped.bam -fq /dev/stdout -fq2 /dev/stdout | pigz > ../../02.unmapped2Cfi_tophat2_fa/tophatfa_csag_unmapped.fq.gz
```
### 将未比对上的reads比对到Cfi重建基因组上
```bash
tophat2 -p 20 -N 1 --read-gap-length 1 --read-edit-dist 1 ../../re-seq/re_con_genome/tophat_index/Cfi/Cfi_genome_reco_JZG ./t
ophatfa_csag_unmapped.fq.gz
```

### 将比对上的序列转为fasta格式
```bash
samtools fasta accepted_hits.bam > hisat2_umapped2cf
i_mapped.fa
```
### 将比对上的序列比对到黄瓜转录组上
```bash
blastn -query ./tophat_fa_umappe
d2cfi_mapped.fa -db ../../rna-seq/03.rmdup.balstndb/Csa/CsaCsa_rmdup_rnaseq -out tophat_fa_umapped2cfi_mapped2csacsa_rnaseq.blastn.xml -outfmt 5 -num_threads 40
```


## 利hisat2比对得后续，利用的fastq文件进行比对

### 利用blastn将比对上的Reads比对到黄瓜的转录组上
```bash
blastn -query ../02.unmapped2Cfi_hisat2/Cfi_rec_mapped.fa -db ../../rna-seq/03.rmdup.balstndb/Csa/CsaCsa_rmdup_rnaseq -out Cfi_rec_mapped2csacsa_rnaseq.blastn.out -outfmt 6 -num_threads 40 -max_target_seqs 10

# 不要把hits（-max_target_seqs）设置的太高，不然输出文件会很大，影响速度和浪费磁盘空间
```

### 将不完全匹配的序列提取出来
```bash
python tongyi_blstn_xml_fuan-jingzange/analysis/03.mapped2csacsarnaseq_tophat_fa/tophat_fa_umapped2cfi_mahisat2/Cfi_rec_mapped2csacsa_rnaseq.blastn.out  hisat2_fq_idlist.txt
```
