<!--
 * @Author: albertxin albert_xin@qq.com
 * @Date: 2024-07-25 08:45:13
 * @LastEditors: albertxin albert_xin@qq.com
 * @LastEditTime: 2024-07-26 09:35:23
 * @FilePath: /shixinblog/blog/2024-07-25-记录terminal.md
 * Copyright (c) albertxin by albert_xin@qq.com, All Rights Reserved. 
-->

# 记录terminal 2024-07-25 08:45:18

## 利用bcftools进行consensus序列的生成

```bash
bcftools consensus -i 'TYPE="snp"' -f ../genome/Cfi_genome_from_JZG.fa Cfi_reseq.vcf.gz -o Cfi_genome_reco_JZG.fa &> bcftools_consensus_snp.log
bcftools consensus -i 'TYPE="snp"' -f ../genome/Cucumber_genome_CLv4.fa Csa_reseq.vcf.gz -o Cucumber_genome_reco.fa &> bcftools_consensus_snp.log
```

### 利用seqkit 进行序列的过滤

```bash
seqkit seq -g -m 100 ../../raw_data/rna-seq/CsaCfi/JZ24116658-GPR240712-10MGI-CsaCfi_combined_R1.fastq.gz |pigz -p 20 > ./CsaCfi_rnaseq_R1.fq.gz
seqkit seq -g -m 100 ../../raw_data/rna-seq/CsaCfi/JZ24116658-GPR240712-10MGI-CsaCfi_combined_R2.fastq.gz |pigz -p 20 > ./CsaCfi_rnaseq_R2.fq.gz

seqkit seq -g -m 100 ../../raw_data/rna-seq/CsaCsa/JZ24116658-GPR240712-10MGI-CsaCsa_combined_R1.fastq.gz |pigz >CsaCsa_rnaseq_R1.fq.gz
seqkit seq -g -m 100 ../../raw_data/rna-seq/CsaCsa/JZ24116658-GPR240712-10MGI-CsaCsa_combined_R2.fastq.gz |pigz >CsaCsa_rnaseq_R2.fq.gz
```


## 利用hisat2比对
### 利用hisat2对从新构建的基因组建立索引

```bash
gffread ./Cfi_genome_from_JZG.gff3 -T -o Cfi_genome_from_JZG.gtf && hisat2_extract_exons.py  Cfi_genome_from_JZG.gtf>  ./Cfi_genome_reco_JZG.exon && hisat2_extract_splice_sites.py  Cfi_genome_from_JZG.gtf >  ./Cfi_genome_reco_JZG.ss && hisat2-build -p 20 --ss ./Cfi_genome_reco_JZG.ss --exon ./Cfi_genome_reco_JZG.exon  ./Cfi_genome_reco_JZG.fa Cfi_genome_reco_JZG.fa  &> hisat2-build.log && samtools faidx Cfi_genome_reco_JZG.fa

gffread ./Cucumber_genome_CLv4.gff3  -T -o Cucumber_genome_CLv4.gtf && hisat2_extract_exons.py  Cucumber_genome_CLv4.gtf >  ./Cucumber_genome_reco.exon && hisat2_extract_splice_sites.py  Cucumber_genome_CLv4.gtf >  ./Cucumber_genome_reco.ss && hisat2-build -p 20 --ss ./Cucumber_genome_reco.ss --exon ./Cucumber_genome_reco.exon  ./Cucumber_genome_reco.fa Cucumber_genome_reco.fa samtools faidx Cucumber_genome_reco.fa &> hisat2-build.log && samtools faidx Cucumber_genome_reco.fa
```



### 利用hisat2 将异源嫁接转录组数据比对到同源嫁接的黄瓜上

```bash
hisat2 -p 20 -x ../../re-seq/re_con_genome/Csa/Cucumber_genome_reco.fa -1 ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz -2 ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R2.fq.gz | samtools sort -O bam -@ 5 -o CsaCfi_rna2Csa-rec_hisat2.sort.bam

```

### 提取未必对上的序列

```bash
samtools view -b -f 4 CsaCfi_rna2Csa-rec_hisat2.sort.bam -O bam | bedtools bamtofastq -i - -fq CsaCfi_rna2Csa-rec_hisat2_fq_unmapped.fq

```

### 将没有比对上的read比对到cfi上

```bash
hisat2 -p 20 -x ../../re-seq/re_con_genome/Cfi/Cfi_genome_reco_JZG.fa -U ../01.Csacfi2Csa_rcg_hisat2/CsaCfi_rna2Csa-rec_hisat2_fq_unmapped.fq.gz  | samtools sort -O bam -@ 5 -o hisat2_unmapped2Cfi_rec.sort.bam
```

### 提取比对上的reads

```bash
samtools view -b -F 4 ./hisat2_unmapped2Cfi_rec.sort.bam -O bam | bedtools bamtofastq -i - -fq /dev/stdout |pigz > Cfi_rec_mapped.fq.gz
```
### 将比对上的reads转化为fasta格式


```bash
seqkit fq2fa Cfi_rec_mapped.fq.gz |pigz >Cfi_rec_mapped.fa.gz
```
### 利用blastn 进行序列比对

```bash

blastn -query ../02.unmapped2Cfi_hisat2/Cfi_rec_mapped.fa.gz -db ../../rna-seq/03.blastndb/Csa/CsaCsa_rnaseq -out Cfi_rec_mapped2csacsa_rnaseq.blastn.out -outfmt 6 -num_threads 20
```


## 利用tophat2比对fastq格式

### 利用Bowtie2建立索引

```bash
bowtie2-build --threads 10 Cfi_genome_reco_JZG.fa Cfi_genome_reco_JZG
bowtie2-build --threads 20 -f Cucumber_genome_reco.fa Cucumber_genome_reco
```
### 利用tophat2 异源嫁接的转录组数据比对到黄瓜上

```bash
tophat2 -p 20 --read-edit-dist 3 ../../re-seq/re_con_genome/tophat_index/Csa/Cucumber_genome_reco  ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R1.fq.gz ../../rna-seq/02.fastqfilter/CsaCfi_rnaseq_R2.fq.gz |samtools sort -O bam -@ 10 -o CsaCfi_rna2Csa-rec.sort.bam

```

### 提取未必对上的序列

```bash
samtools view -b -f 4 CsaCfi_rna2Csa-rec.sort.bam -O bam | bedtools bamtofastq -i - -fq CsaCfi_rna2Csa-rec_unmapped_R1.fq -fq2 CsaCfi_rna2Csa-rec_unmapped_R2.fq

```

## 利用tophat2比对fasta格式的转录组序列


## 利用fasta比对
```bash
tophat2 -p 20 --b2-f  --read-edit-dist 3 ../../re-seq/re_con_genome/tophat_index/Csa/Cucumber_genome_reco ../../rna-seq/01-merged-fa/CsaCfi_100bp.fa.gz |samtools sort -O bam -@ 10 -o CsaCfi_rna2Csa-rec_tophat2_fa.sort.bam
```
